### DISCUSSION TITLE
Is it possible to set `user` and `user-group` in `exec` ?

### DISCUSSION THREAD

**Discussion by Seasquash** [kube-rs/kube#937] [Q&A] [answered]
Hi there.
I'm trying to execute a command in a pod but have to use a different user/group.
The corresponding command using `kubectl` would be something like:
```
kubectl exec --as xxx --as-group xxx:xxx ...
```

At the moment, I can access the pod and run the command but fails with a `403`:
```
let pods_api: Api<Pod> = Api::namespaced(client, "xxxx".into());
let attached = pods_api.exec(&pod_name, command, &AttachParams {
        container: Some("xxxxxx".into()),
        stderr: false,
        ..Default::default()
    }).await?;
```

Any hint?

Thanks!

**Comment by clux** [comment, marked as answer]
Hey there, the [impersonation stuff](https://github.com/kube-rs/kube-rs/pull/797) does go into the kubeconfig normally. To do it programmatically you need to override the `AuthInfo` struct either in `Config` or in `ConfigLoader`.

It is also possible to use some of our debug variables for quick tests:
```sh
KUBE_RS_DEBUG_IMPERSONATE_USER=xxx KUBE_RS_DEBUG_IMPERSONATE_GROUP=xxx cargo run
```

**Reply by Seasquash** [reply]
Ahhh, fair point... I'll try with `AuthInfo`, cause I did have a go at setting those env variables but with no luck ...
I'm wondering if there's more stuff under the hood that needs wiring up on my side, since the role is handled by Okta, but that's a different story.. 😝 
Thanks Erik!

**Reply by Seasquash** [reply]
Just dropping a wee note: thanks for pointing me in the right direction! Dug into Kubernetes API a bit more and managed to read from the file config and override the AuthInfo with my impersonation properties.
